### PR TITLE
#157891041 Resolves bugs in admin approve and resolve endpoints

### PR DESCRIPTION
**Pull request by chukwuemekachm**
#### What does this PR do?
- Resolves bugs in admin approve, disapprove and resolve endpoints
#### Description of Task to be completed?
- Write tests for endpoints
- Modidy api/v1/requests/:requestId/approve
- Modidy api/v1/requests/:requestId/disapprove
- Modidy api/v1/requests/:requestId/resolve
- Test endpoint using Postman
- Enable cors on server
#### How should this be manually tested?
- Clone the repo
- Startup command line
- cd to project root
- run ``` npm install ```
- run ``` npm run migrate ```
- run ``` npm run start:dev ```
- Test endpoint using Postman
#### Any background context you want to provide?
- Add a ``` .env ``` file to the project root
- In the ``` .env ``` file set the DATABASE_URL to the `URL` your preferred database
#### What are the relevant pivotal tracker stories?
- #157891041
#### Screenshots (if appropriate)

screenshot 192

#### Questions:
- None

### CODE REVIEW

**Comment by benfluleck** [comment]
I don't need to see screenshots of your code it does not contribute anything special to this PR. Please have a body for your commit messages you can find the convention pinned to the top of the slack channel.

File: Server/middlewares/checkRequest.js [hunk 1]
```diff
@@ -23,14 +23,49 @@ const requestCheck = (req, res, next) => {
...
+  };
+  const client = new Client({
+    connectionString,
+  });
+  client.connect();
+  client.query(queryString, (error, result) => {
+    client.end();
+    if (!result.rows[0]) { return res.status(404).json({ status: 'error', code: 404, message: 'Request does not exist' }); }
+    const { status } = result.rows[0];
+    if (status === 'approved') {
```
Comment (benfluleck) [inline Server/middlewares/checkRequest.js:61]: **Nitpick**
Is there a benefit by setting the status to approved as opposed to using a number for the status codes e.g. 1 for approved, 2 for rejected. I feel using this way it's a quicker process and it is not as tightly coupled.

File: Server/middlewares/validation.js [hunk 2]
```diff
@@ -102,7 +102,7 @@ const signup = (req, res, done) => {
   if (!lastname || typeof lastname !== 'string' || !/^[a-zA-Z]{1,}$/.test(lastname)) {
     return res.status(400).json({ status: 'error', code: 400, message: 'lastname is required or invalid' });
   }
-  if (!email || typeof email !== 'string') {
+  if (!email || typeof email !== 'string' || !/^[a-zA-Z0-9]+@[a-zA-Z0-9]+\.[A-Za-z]{2,4}$/.test(email)) {
```
Comment (benfluleck) [inline Server/middlewares/validation.js:105]: Accolades for using the joi package to achieve this effect 👍

File: package.json [hunk 3]
```diff
@@ -25,14 +25,15 @@
...
+        "dotenv": "^5.0.1",
         "errorhandler": "^1.5.0",
         "express": "^4.16.3",
         "joi": "^13.3.0",
         "jsonwebtoken": "^8.2.1",
         "morgan": "^1.9.0",
         "pg": "^7.0.3",
-        "winston": "^2.4.2",
-        "dotenv": "^5.0.1"
+        "winston": "^2.4.2"
```
Comment (benfluleck) [inline package.json]: Where are you using this package?

File: Server/middlewares/checkRequest.js (hunk 1 above)
Comment (chukwuemekachm) [reply Server/middlewares/checkRequest.js:61]: The numbering process is quicker, but when I retrieve the information from the database I would have to do checks and conversions on client or server in other to make it understandable to a person using the API.

File: package.json (hunk 3 above)
Comment (chukwuemekachm) [reply package.json]: I used it previously but forgot to uninstall it

**Pull request merged**